Toggl.Foundation/TogglSyncManagerFactory.cs

```
@@ -210,17 +230,15 @@ public static class TogglSyncManager
             transitions.ConfigureTransition(noDefaultWorkspaceDetectingState.NoDefaultWorkspaceDetected, trySetDefaultWorkspaceState);
 
             // process server errors
-            transitions.ConfigureTransition(persistWorkspaces.ErrorOccured, retryOrThrow);
-            transitions.ConfigureTransition(persistUser.ErrorOccured, retryOrThrow);
-            transitions.ConfigureTransition(persistWorkspaceFeatures.ErrorOccured, retryOrThrow);
-            transitions.ConfigureTransition(persistPreferences.ErrorOccured, retryOrThrow);
-            transitions.ConfigureTransition(persistTags.ErrorOccured, retryOrThrow);
-            transitions.ConfigureTransition(persistClients.ErrorOccured, retryOrThrow);
-            transitions.ConfigureTransition(persistProjects.ErrorOccured, retryOrThrow);
-            transitions.ConfigureTransition(persistTasks.ErrorOccured, retryOrThrow);
-            transitions.ConfigureTransition(createProjectPlaceholders.ErrorOccured, retryOrThrow);
-            transitions.ConfigureTransition(persistTimeEntries.ErrorOccured, retryOrThrow);
-            transitions.ConfigureTransition(refetchInaccessibleProjects.ErrorOccured, retryOrThrow);
```

😱 I summon @amulware. This might cause an infinite syncing loop: when refetching of the project fails (e.g., the user doesn't have access to the workspace), the server returns 4xx, we loop back to `fetchAllSince` and it is guaranteed that it will fail again.

Maybe we should just silently skip the projects which we can't refetch?

---

This PR should probably just keep things as they were and we should create an issue for this and fix it. And create a sync test for it.